release/src/router/rom/webs_scripts/merlin_webs_update.sh

```diff
@@ -18,7 +18,7 @@ fi
 current_base=$(nvram get firmver | sed "s/\.//g")
 current_firm=$(nvram get buildno | cut -d. -f1)
 current_buildno=$(nvram get buildno | cut -d. -f2)
-current_extendno=$(nvram get extendno | sed "s/-g.*//" | sed "s/_.*//" | sed "s/alpha\|beta/-1/")
+current_extendno=$(nvram get extendno | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}') #Extract extendno, subtract value by 1 if it contains "alpha/beta", remove all other values such as "_rog" or "-g*"
```

Comment:
This has never properly worked...
# Current solution:

Define the extendnos as a space-separated string:
```
extendnos="2_alpha1-g4889cee45a 0 2_alpha3 3 0_rog 0_alpha1 0_rog_alpha1 1_alpha1 1_rog_alpha1 0_Alpha1 0_Rog_Alpha1 0_Beta2 1_beta3"
```
Loop through each extendno:
```
for extendno in $extendnos; do
  current_extendno=$(echo "$extendno" | sed "s/-g.*//" | sed "s/_.*//" | sed "s/alpha\|beta/-1/")
  echo "Original: $extendno -> Processed: $current_extendno"
done
```

# Output:
Original: 2_alpha1-g4889cee45a -> Processed: 2
Original: 0 -> Processed: 0
Original: 2_alpha3 -> Processed: 2
Original: 3 -> Processed: 3
Original: 0_rog -> Processed: 0
Original: 0_alpha1 -> Processed: 0
Original: 0_rog_alpha1 -> Processed: 0
Original: 1_alpha1 -> Processed: 1
Original: 1_rog_alpha1 -> Processed: 1
Original: 0_Alpha1 -> Processed: 0
Original: 0_Rog_Alpha1 -> Processed: 0
Original: 0_Beta2 -> Processed: 0
Original: 1_beta3 -> Processed: 1

Reply:
# Proposed solution:

Define the extendnos as a space-separated string
```
extendnos="2_alpha1-g4889cee45a 0 2_alpha3 3 0_rog 0_alpha1 0_rog_alpha1 1_alpha1 1_rog_alpha1 0_Alpha1 0_Rog_Alpha1 0_Beta2 1_beta3"
```

Loop through each extendno
```
for extendno in $extendnos; do
  # Process each extendno in a single line
  processed_number=$(echo "$extendno" | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}')
  echo "Original: $extendno -> Processed: $processed_number"
done
```

# Output:
Original: 2_alpha1-g4889cee45a -> Processed: 1
Original: 0 -> Processed: 0
Original: 2_alpha3 -> Processed: 1
Original: 3 -> Processed: 3
Original: 0_rog -> Processed: 0
Original: 0_alpha1 -> Processed: -1
Original: 0_rog_alpha1 -> Processed: -1
Original: 1_alpha1 -> Processed: 0
Original: 1_rog_alpha1 -> Processed: 0
Original: 0_Alpha1 -> Processed: -1
Original: 0_Rog_Alpha1 -> Processed: -1
Original: 0_Beta2 -> Processed: -1
Original: 1_beta3 -> Processed: 0